### PR TITLE
Implementing Contrastive Self-Supervised Learning with Radiation Augmentations, SimCLR, PyTorch Lightning, and Hyperparameter Optimization

### PR DESCRIPTION
**Pull request by stompsjo**
This PR constitutes the bulk of my code (excluding notebooks used to generate and analyze results) used in my dissertation. Some highlights:
- Augmentations for gamma radiation spectra (`augs.py`) including a way to use them in contrastive learning (`transforms.py`).
- Basic infrastructure for creating a PyTorch/Lightning (`ann.py`/`lightModel.py`) multilayer perceptron (`LinearNN`) and convolutional neural network (`ConvNN`), and a projection head (`critic.py`).
- A self-/semi-supervised contrastive learning training script written for both PyTorch (`SlimCLR.py`) and PyTorch Lightning (`SlimCLRLight.py`) using the package `pytorch-learning-metrics` for a normalized cross-entropy loss function based on SimCLR.
- Scripts for conducting hyperparameter optimization on the base encoder (`SSLHyperOpt.py`) and the projection head (`ProjHyperOpt.py`) using the package `hyperopt`.
- A snapshot of the conda/python environment I used for my research (`contrastive-environment.yml`).

There is _a lot_ of code in this branch, so it will undoubtedly make sense to split it up. I also need to clean up scripts. Many include functions that I ended up not using, or were borrowed from other people (e.g. `specTools.py` from Ken) and require proper attribution. If we want to move to review and merge this PR, we should probably finish reviewing and merging the preliminary work PRs in #42, #44, #45, #46.

- [ ] UNIT TESTS

### CODE REVIEW

**Comment by stompsjo** [comment]
Closing in favor of #52.
